orchagent/portsorch.cpp

```diff
@@ -345,6 +345,74 @@ bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
     return true;
 }
 
+bool PortsOrch::setPortPvid (Port &port, sai_uint32_t pvid)
+{
+    SWSS_LOG_ENTER();
+    vector<Port> portv;
+
+    if (port.m_type == Port::PHY)
+    {
+        portv.push_back(port);
+    }
+    else if (port.m_type == Port::LAG)
+    {
+        getLagMember(port, portv);
+    }
+    else
+    {
+        SWSS_LOG_ERROR("PortsOrch::setPortPvid port type %d not supported", port.m_type);
+        return false;
+    }
+
+    for (const auto p: portv)
+    {
+        sai_attribute_t attr;
+        attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+        attr.value.u32 = pvid;
+
+        sai_status_t status = sai_port_api->set_port_attribute(p.m_port_id, &attr);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set pvid %u to port: %s", attr.value.u32, p.m_alias.c_str());
+            return false;
+        }
+        SWSS_LOG_NOTICE("Set pvid %u to port: %s", attr.value.u32, p.m_alias.c_str());
+    }
+    return true;
+}
+
+bool PortsOrch::getPortPvid(Port &port, sai_uint32_t &pvid)
+{
+    /* Just return false if the router interface exists */
+    if (port.m_rif_id)
+    {
+        SWSS_LOG_DEBUG("Router interface exists on %s, don't set pvid",
+                      port.m_alias.c_str());
+        return false;
+    }
+
+    if (port.m_port_vlan_id != DEFAULT_PORT_VLAN_ID)
+    {
+        pvid = port.m_port_vlan_id;
+        return true;
+    }
+
+    if (port.m_vlan_members.size() == 0)
+    {
+        pvid = DEFAULT_PORT_VLAN_ID;
+        return true;
+    }
+
+    for (auto &vme: port.m_vlan_members)
+    {
+        if(vme.second.vlan_mode == SAI_VLAN_TAGGING_MODE_UNTAGGED) {
+            pvid =  vme.first;
+            return true;
+        }
+    }
+    pvid = DEFAULT_PORT_VLAN_ID;
```

Comment:
this is complicated, when the port is not a router interface, we should just return port.m_port_vlan_id, and we should make the m_port_vlan_id correct.

Reply:
As explained previously, it was for supporting pvid configuration from admin interface.

Updated.